docs/persistence.md

```diff
@@ -0,0 +1,110 @@
+# Persistence
+
+Generally speaking, adding database storage or persistence to your Probot App will greatly complicate your life. It's perfectly doable, but for most use-cases you'll be able to manage relevant data within GitHub issues, pull requests and your app's configuration file.
```

Comment:
It'd be cool to have a section on tips for using GitHub as the source of truth. Here are a few examples that might be worth calling out:

- **Comments:** - [snooze](https://github.com/jbjonesjr/probot-snooze) embeds JSON inside HTML comments (`<!-- {"json": "here"} -->`) in comments posted by the app.
- **Statuses:** [DCO](https://probot.github.io/apps/dco/) uses the [statuses](https://developer.github.com/v3/repos/statuses/) API
- **Repository:** Built-in `context.config()` allows storing [configuration in the repository](https://probot.github.io/docs/best-practices/#store-configuration-in-the-repository)
- **Search:** - [first-per-merge](https://github.com/behaviorbot/first-pr-merge) uses search to see if a user has had a PR merged before now
- **Labels:** - [stale](https://github.com/probot/stale) uses search to find issues and labels to manage state
- **Misc APIs** - [scheduler](https://github.com/probot/scheduler/blob/master/index.js#L69) fetches the [list of installations](https://github.com/probot/scheduler/blob/master/index.js#L69) from the API

I'm sure there are other examples.

Reply:
Good idea. I'll link out to those individual APIs as documented in `node-github` as well.

Reply:
It's not public yet, but I have a project that uses `.yml` files to store links shared by a community. Jekyll can use these `.yml` files to either render HTML views or export something like an API, it’s pretty straight forward.

Reason why I specifically decided to use GitHub itself as the database is because

1. GitHub has a built-in review process and great settings to enforce workflows
2. Attribution. I know who contributed what, and who reviewed a submission
3. It integrates with other GitHub features, e.g. blocking users
4. And yeah, I don’t need to run and maintain a database somewhere :D